db/log_reader.cc

```diff
@@ -700,9 +744,33 @@ bool FragmentBufferedReader::TryReadFragment(
 
   buffer_.remove_prefix(header_size + length);
 
-  *fragment = Slice(header + header_size, length);
-  *fragment_type_or_err = type;
-  return true;
+  if (uncompress_ && type != kSetCompressionType) {
```

Comment:
It's possible that re-arranging this `if` (negating the condition and swapping the "then" and "else" code blocks) could negate some of the regression. Why? Generally default branch predictions would be not to jump forward (favoring "then" rather than "else"). And possibly better code locality because of "return" from "then" and "else". You can also put `LIKELY()` around `type != kSetCompressionType`.

Reply:
That seemed to help. I'll run it one more time to be sure.

Without PR -
Avg ops/sec 478815

fillseq      :       2.042 micros/op 489809 ops/sec;   54.2 MB/s
fillseq      :       2.134 micros/op 468617 ops/sec;   51.8 MB/s
fillseq      :       2.066 micros/op 484099 ops/sec;   53.6 MB/s
fillseq      :       2.140 micros/op 467203 ops/sec;   51.7 MB/s
fillseq      :       2.146 micros/op 465934 ops/sec;   51.5 MB/s
fillseq      :       2.038 micros/op 490768 ops/sec;   54.3 MB/s
fillseq      :       2.194 micros/op 455855 ops/sec;   50.4 MB/s
fillseq      :       2.051 micros/op 487570 ops/sec;   53.9 MB/s
fillseq      :       2.075 micros/op 481840 ops/sec;   53.3 MB/s
fillseq      :       2.150 micros/op 465151 ops/sec;   51.5 MB/s
fillseq      :       2.062 micros/op 484875 ops/sec;   53.6 MB/s
fillseq      :       2.054 micros/op 486925 ops/sec;   53.9 MB/s
fillseq      :       2.095 micros/op 477330 ops/sec;   52.8 MB/s
fillseq      :       2.063 micros/op 484727 ops/sec;   53.6 MB/s
fillseq      :       2.020 micros/op 495089 ops/sec;   54.8 MB/s
fillseq      :       2.093 micros/op 477879 ops/sec;   52.9 MB/s
fillseq      :       2.133 micros/op 468929 ops/sec;   51.9 MB/s
fillseq      :       2.014 micros/op 496462 ops/sec;   54.9 MB/s
fillseq      :       2.151 micros/op 464845 ops/sec;   51.4 MB/s
fillseq      :       2.073 micros/op 482397 ops/sec;   53.4 MB/s

With PR -
Avg ops/sec 470565

fillseq      :       2.162 micros/op 462585 ops/sec;   51.2 MB/s
fillseq      :       2.092 micros/op 478039 ops/sec;   52.9 MB/s
fillseq      :       2.103 micros/op 475427 ops/sec;   52.6 MB/s
fillseq      :       2.081 micros/op 480608 ops/sec;   53.2 MB/s
fillseq      :       2.147 micros/op 465735 ops/sec;   51.5 MB/s
fillseq      :       2.117 micros/op 472425 ops/sec;   52.3 MB/s
fillseq      :       2.092 micros/op 478105 ops/sec;   52.9 MB/s
fillseq      :       2.117 micros/op 472352 ops/sec;   52.3 MB/s
fillseq      :       2.116 micros/op 472519 ops/sec;   52.3 MB/s
fillseq      :       2.131 micros/op 469370 ops/sec;   51.9 MB/s
fillseq      :       2.154 micros/op 464144 ops/sec;   51.3 MB/s
fillseq      :       2.098 micros/op 476660 ops/sec;   52.7 MB/s
fillseq      :       2.139 micros/op 467601 ops/sec;   51.7 MB/s
fillseq      :       2.170 micros/op 460933 ops/sec;   51.0 MB/s
fillseq      :       2.124 micros/op 470743 ops/sec;   52.1 MB/s
fillseq      :       2.120 micros/op 471664 ops/sec;   52.2 MB/s
fillseq      :       2.115 micros/op 472834 ops/sec;   52.3 MB/s
fillseq      :       2.174 micros/op 460034 ops/sec;   50.9 MB/s
fillseq      :       2.123 micros/op 471077 ops/sec;   52.1 MB/s
fillseq      :       2.135 micros/op 468455 ops/sec;   51.8 MB/s

Reply:
Subsequent runs, I still see some regression -
490107 vs 473037 ops/sec

Reply:
I think its noise. The variation without your changes (478815 vs 490107) suggests so. Also, PGO should take care of rearrangement if necessary?

Reply:
Ok. Let me submit a diff.